File: step04customlist/src/main/java/com/gura/step04customlist/DetailActivity.java
package com.gura.step04customlist;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import org.w3c.dom.Text;

import java.io.Serializable;

public class DetailActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        //DetailActivity 가 활성화 되는데 사용되었던 Intent 객체의 참조값 얻어오기
        Intent intent=getIntent();
        // "dto"라는 키값으로 전달된 Serializable type 객체의 참조값 얻어오기
        Serializable serial=intent.getSerializableExtra("dto");
        //원래 type으로 casting!
        CountryDto dto=(CountryDto)serial;

        /*
            위의 코드를 한줄로 적으면 아래와 같다.
            CountryDto dto=(CountryDto)getIntent().getSerializableExtra("dto");
         */

        // 전개된 레이아웃에서 UI 참조값 얻어오기
        ImageView imageView=findViewById(R.id.imageView);
        TextView textView=findViewById(R.id.textView);
        Button confirmBtn=findViewById(R.id.confirmBtn);

        // 이미지와 내용 출력하기
        imageView.setImageResource(dto.getResId());
        textView.setText(dto.getContent());

        //버튼에 리스너 등록하기
        confirmBtn.setOnClickListener(this);
    }


    @Override
    public void onClick(View v) {
        //엑티비티 종료 시키기
        finish();
    }
}

File: step22notification/src/main/java/com/gura/step22notification/DetailActivity.kt
package com.gura.step22notification

import android.app.NotificationManager
import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_detail.*

class DetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

    }

    override fun onStart() {
        super.onStart()

        //Intent 객체에 "msg" 라는 키값으로 담긴 문자열 읽어오기
        var msg=intent.getStringExtra("msg") //getIntent() 대신에 kotlin에서는 intent 라고 작성해주면 된다.
        System.out.println(msg)
        //TextView 에 출력하기
        textView.setText(msg) //findById() 대신에 kotlin에서는 textView 라고 작성해주면 된다.
        //Intent 객체에 "id" 라는 키값으로 전달된 Int 값이 있는지 읽어와 본다.
        var id=intent.getIntExtra("id", -1)
        if(id!=-1){ //알림을 수동을 취소 해야하는 경우
            //알림 매니저 객체의 참조값을 얻어와서
            val manager=getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            //알림 id 값을 전달해서 취소 시킨다.
            //manager.cancel(id)
            manager.cancelAll() //: 모든 알림을 취소 시킨다.
        }
    }
}
